ext/OceananigansMakieExt.jl

```diff
@@ -0,0 +1,57 @@
+module OceananigansMakieExt
+
+using Oceananigans
+using Oceananigans.Architectures: on_architecture
+
+using MakieCore: AbstractPlot
+import MakieCore: convert_arguments
+
+function drop_singleton_indices(N)
+    if N == 1
+        return 1
+    else
+        return Colon()
+    end
+end
+
+function convert_arguments(pl::Type{<:AbstractPlot}, f::Field)
+
+    # Deduce dimensionality
+    Nx, Ny, Nz = size(f)
```

Comment:
so are you implying that f should be a 2D slice?
if so, how would this work... size would return a 2-tuple...

Reply:
The code I wrote works --- its not a concept! `size` is always a 3-tuple:

```julia
julia> size(c)
(10, 10, 10)

julia> size(view(c, :, 2, :))
(10, 1, 10)
```

Reply:
```julia
julia> r = Field{Center, Center, Nothing}(grid)
si10×10×1 Field{Center, Center, Nothing} reduced over dims = (3,) on RectilinearGrid on CPU
├── grid: 10×10×10 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
├── boundary conditions: FieldBoundaryConditions
│   └── west: Periodic, east: Periodic, south: Periodic, north: Periodic, bottom: Nothing, top: Nothing, immersed: ZeroFlux
└── data: 16×16×1 OffsetArray(::Array{Float64, 3}, -2:13, -2:13, 1:1) with eltype Float64 with indices -2:13×-2:13×1:1
    └── max=0.0, min=0.0, mean=0.0

julia> size(r)
(10, 10, 1)

julia> grid = RectilinearGrid(size=(10, 10), x=(0, 1), y=(0, 1), topology=(Periodic, Periodic, Flat))
10×10×1 RectilinearGrid{Float64, Periodic, Periodic, Flat} on CPU with 3×3×0 halo
├── Periodic x ∈ [0.0, 1.0) regularly spaced with Δx=0.1
├── Periodic y ∈ [0.0, 1.0) regularly spaced with Δy=0.1
└── Flat z

julia> c = CenterField(grid)
si10×10×1 Field{Center, Center, Center} on RectilinearGrid on CPU
├── grid: 10×10×1 RectilinearGrid{Float64, Periodic, Periodic, Flat} on CPU with 3×3×0 halo
├── boundary conditions: FieldBoundaryConditions
│   └── west: Periodic, east: Periodic, south: Periodic, north: Periodic, bottom: Nothing, top: Nothing, immersed: ZeroFlux
└── data: 16×16×1 OffsetArray(::Array{Float64, 3}, -2:13, -2:13, 1:1) with eltype Float64 with indices -2:13×-2:13×1:1
    └── max=0.0, min=0.0, mean=0.0

julia> size(c)
(10, 10, 1)

Reply:
Gotcha! I was thinking of 

```Julia
julia> size(interior(c, :, :, 1))
(10, 10)
```

Reply:
Right, `interior` returns a `SubArray` but this is just for fields.

Reply:
My bad though since `interior(c, :, :, 1)` is _not_ a field!